### ISSUE TITLE
[Bug] Missing extensions on services

### ISSUE THREAD

**Issue by giranm** [giranm/pd-live-react#138]
## Summary
For subdomains with large number of services, the UI fails to accurately map all extensions associated with those services.  
The implication is that users do not see the extension (e.g. ServiceNow) when trying to run an action on a selected incident.  

To remedy this `mapServicesToExtensionsImpl()` needs a wait condition for all services to be completed before the mapping is done.

#### Suggested Fix
```javascript
export function* mapServicesToExtensionsImpl() {
  try {
    // Wait until services have been fetched before mapping extensions
    yield take([FETCH_SERVICES_COMPLETED, FETCH_SERVICES_ERROR]);
    // Remainder of code
```

**Comment by giranm** [comment]
While attempting to fix this bug, I noticed different behaviours across different subdomains, mostly due to the number of services and extensions compared to our integration environment. Smaller subdomains did not have the race condition that larger domains did, hence the difficulty with identifying this bug.

Due to these differences, the most logical approach is to make the fetches in a linear fashion, starting with obtaining the services, then extensions, and finally doing the map. This meant that the mapping was consistent for all subdomains.

**Comment by giranm** [comment]
Merged #187 - closing.
